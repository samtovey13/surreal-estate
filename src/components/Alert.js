import React from "react";
import "../styles/Alert.css";
import PropTypes from "prop-types";

const Alert = ({ message, success }) => {
  const green = { text: "#007a33", border: "#d3edd9" };
  const red = { text: "#de0000", border: "#ffc2c2" };
  const color = success ? green : red;

  if (!message) return null;
  
  return (
    <div className="alert-container">
      <div
        className="alert"
        data-testid="alert"
        style={{
          color: color.text,
          borderColor: color.border,
        }}
      >
        {message}
      </div>
    </div>
  );
};

Alert.propTypes = {
  message: PropTypes.string.isRequired,
  success: PropTypes.bool,
};

export default Alert;
