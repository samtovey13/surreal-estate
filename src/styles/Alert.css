.alert {
  border: 5px solid;
  padding: 10px;
  max-width: 600px;
  margin: 15px;
  box-sizing: border-box;
  border-radius: 3px;
  background-color: white;
}