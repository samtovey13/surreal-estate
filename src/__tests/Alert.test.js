import { render } from "@testing-library/react";
import Alert from "../components/Alert";

describe("Alert", () => {
  it("renders a success message", () => {
    const { getByText, asFragment } = render(
      <Alert message="Success message" success={true} />
    );

    expect(asFragment()).toMatchSnapshot();
    expect(getByText("Success message")).toHaveClass("alert");
  });

  it("renders an error message", () => {
    const { getByText, asFragment } = render(
      <Alert message="Error message" success={false} />
    );

    expect(asFragment()).toMatchSnapshot();
    expect(getByText("Error message")).toHaveClass("alert");
  });

  it("does not render a message when message is empty", () => {
    const { asFragment, queryByTestId } = render(
      <Alert message="" success={false} />
    );

    expect(asFragment()).toMatchSnapshot();
    expect(queryByTestId("alert")).not.toBeInTheDocument();
  });
});
