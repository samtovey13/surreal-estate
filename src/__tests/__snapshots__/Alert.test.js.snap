// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert does not render a message when message is empty 1`] = `
<DocumentFragment>
  <div
    class="alert"
    style="color: rgb(222, 0, 0); border-color: #ffc2c2;"
  />
</DocumentFragment>
`;

exports[`Alert renders a success message 1`] = `
<DocumentFragment>
  <div
    class="alert"
    style="color: rgb(0, 122, 51); border-color: #d3edd9;"
  >
    Success message
  </div>
</DocumentFragment>
`;

exports[`Alert renders an error message 1`] = `
<DocumentFragment>
  <div
    class="alert"
    style="color: rgb(222, 0, 0); border-color: #ffc2c2;"
  >
    Error message
  </div>
</DocumentFragment>
`;
